handlers/prometheus_handlers.go

```diff
@@ -1,282 +1,308 @@
-	if req.Method != http.MethodGet {
-		w.WriteHeader(http.StatusNotFound)
-		return
-	}
-
-	if prefObj.Prometheus == nil || prefObj.Prometheus.PrometheusURL == "" {
-		_, _ = w.Write([]byte("{}"))
-		return
-	}
-
-	result := map[string]*models.GrafanaBoard{}
-	resultLock := &sync.Mutex{}
-	resultWG := &sync.WaitGroup{}
-
-	boardFunc := map[string]func(context.Context, string) (*models.GrafanaBoard, error){
-		"cluster": h.config.PrometheusClient.GetClusterStaticBoard,
-		"node":    h.config.PrometheusClient.GetNodesStaticBoard,
-	}
-
-	for key, bfunc := range boardFunc {
-		resultWG.Add(1)
-		go func(k string, bfun func(context.Context, string) (*models.GrafanaBoard, error)) {
-			defer resultWG.Done()
-
-			board, err := bfun(req.Context(), prefObj.Prometheus.PrometheusURL)
-			if err != nil {
-				// error is already logged
-				return
-			}
-			resultLock.Lock()
-			defer resultLock.Unlock()
-			result[k] = board
-		}(key, bfunc)
-	}
-	resultWG.Wait()
-
-	if len(result) != len(boardFunc) {
-		http.Error(w, "unable to get static board", http.StatusInternalServerError)
-		return
-	}
-
-	err := json.NewEncoder(w).Encode(result)
-	if err != nil {
-		logrus.Errorf("error marshalling board: %v", err)
-		http.Error(w, "unable to marshal board instance", http.StatusInternalServerError)
-		return
-	}
-}
-
-// SaveSelectedPrometheusBoardsHandler persists selected board and panels
-func (h *Handler) SaveSelectedPrometheusBoardsHandler(w http.ResponseWriter, req *http.Request, _ *sessions.Session, prefObj *models.Preference, user *models.User, provider models.Provider) {
-	if req.Method != http.MethodPost {
-		w.WriteHeader(http.StatusNotFound)
-		return
-	}
-
-	if prefObj.Prometheus == nil || prefObj.Prometheus.PrometheusURL == "" {
-		http.Error(w, "Prometheus URL is not configured", http.StatusBadRequest)
-		return
-	}
-
-	// if prefObj.Prometheus.SelectedPrometheusBoardsConfigs == nil {
-	// 	prefObj.Prometheus.SelectedPrometheusBoardsConfigs = []*models.GrafanaBoard{}
-	// }
-
-	defer func() {
-		_ = req.Body.Close()
-	}()
-
-	body, err := ioutil.ReadAll(req.Body)
-	if err != nil {
-		msg := "unable to read the request body"
-		logrus.Error(errors.Wrapf(err, msg))
-		http.Error(w, msg, http.StatusInternalServerError)
-		return
-	}
-	boards := []*models.SelectedGrafanaConfig{}
-	err = json.Unmarshal(body, &boards)
-	if err != nil {
-		msg := "unable to parse the request body"
-		logrus.Error(errors.Wrapf(err, msg))
-		http.Error(w, msg, http.StatusBadRequest)
-		return
-	}
-	if len(boards) > 0 {
-		prefObj.Prometheus.SelectedPrometheusBoardsConfigs = boards
-	} else {
-		prefObj.Prometheus.SelectedPrometheusBoardsConfigs = nil
-	}
-	err = provider.RecordPreferences(req, user.UserID, prefObj)
-	if err != nil {
-		logrus.Errorf("unable to save user config data: %v", err)
-		http.Error(w, "unable to save user config data", http.StatusInternalServerError)
-		return
-	}
-	_, _ = w.Write([]byte("{}"))
-}
+package handlers
+

```

Comment:
@vijcheru i wonder if you might quickly highlight the lines changed in these files. The lack of line-by-line diff highlighting makes it a challenge to pick out what have changed.

Reply:
Agreed @vijcheru , it would be really helpful to review if you could just edit the lines that are affected by this PR.